core/shared/src/main/scala/io/circe/Error.scala

```diff
@@ -20,6 +22,11 @@ final object ParsingFailure {
   implicit final val eqParsingFailure: Eq[ParsingFailure] = Eq.instance {
     case (ParsingFailure(m1, t1), ParsingFailure(m2, t2)) => m1 == m2 && t1 == t2
   }
+
+  implicit final val showParsingFailure: Show[ParsingFailure] = Show.show { failure =>
+    val name = classOf[ParsingFailure].getSimpleName
```

Comment:
What's the argument for getting the name like this? I guess it could help in refactoring, but these names are likely to be pretty stable, and I'm inclined to prefer a blanket prohibition on runtime reflection in circe-core (even though in this case it's clearly not unsafe). I could be convinced the other way, though.